contracts/modules/HealerModule.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.14;
+
+import "@yield-protocol/vault-interfaces/src/ICauldron.sol";
+import "@yield-protocol/vault-interfaces/src/ILadle.sol";
+import "@yield-protocol/utils-v2/contracts/math/WMul.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I128.sol";
+import "../LadleStorage.sol";
+
+///@title Ladle module that allows any vault to be poured to as long as it adds collateral or repays debt
+contract HealerModule is LadleStorage {
+    using WMul for uint256;
+    using CastU256I128 for uint256;
+
+    constructor(ICauldron cauldron_, IWETH9 weth_) LadleStorage(cauldron_, weth_) {}
+
+    function heal(bytes12 vaultId, int128 ink, int128 art)
+        external payable
+    {
+        require (ink >= 0, "Only add collateral");
+        require (art <= 0, "Only repay debt");
+        (, DataTypes.Vault memory vault) = getVault(vaultId);        
+        cauldron.pour(vaultId, ink, art);
+
+        // Manage collateral
+        if (ink != 0) {
+            IJoin ilkJoin = getJoin(vault.ilkId);
+            ilkJoin.join(vault.owner, uint128(ink));
```

Comment:
Actually, this is not right.

[Join.join](https://github.com/yieldprotocol/vault-v2/blob/1425bd260c80c0e3dce2d434172c4bc43d9e1ccf/packages/foundry/contracts/Join.sol#L27) will pull assets from the account passed on as a parameter, if it has an allowance to do so.

We don't usually make users `approve` the Join, but if someone did, then anyone could call this module to pull assets from the user into their vault, up to the allowance.

Hardly a disaster, no funds would be lost, still it is an undesiderable behaviour.

A better option would be:
```suggestion
            ilkJoin.join(msg.sender, uint128(ink));
```
In that case, it is responsability of the caller to decide whether to send funds in a previous call, or approve the Join and let it pull the funds.